torchaudio/csrc/CMakeLists.txt

```diff
@@ -59,45 +112,21 @@ set_target_properties(libtorchaudio PROPERTIES PREFIX "")
 target_include_directories(
   libtorchaudio
   PRIVATE
-  ${PROJECT_SOURCE_DIR}
+  ${LIBTORCHAUDIO_INCLUDE_DIRS}
   )
-
 target_link_libraries(
   libtorchaudio
-  torch
-  ${TORCHAUDIO_THIRD_PARTIES}
+  ${LIBTORCHAUDIO_LINK_LIBRARIES}
+  )
+target_compile_definitions(
+  libtorchaudio
+  PRIVATE ${LIBTORCHAUDIO_COMPILE_DEFINITIONS}
```

Comment:
what are the consequences of making the scope of `INCLUDE_SOX` and `INCLUDE_KALDI` `PRIVATE` rather than `PUBLIC` as before?

Reply:
The direct impact is that, previously `_torchaudio` was inheriting these definitions automatically, so they did not need to have the definition again.

The indirect impact I can think of is that if third party project is using this CMakeLists to integrate `libtorchaudio`, and they customize the build process of their application alongside `libtorchaudio`, using the same compiler definition `INCLUE_SOX` and `INCLUDE_KALDI`,  then they have to add the same compile definition by them selves, as it will be no longer visible.

My justification is that, since the third party library has control over whether to include these dependencies, they have enough information to do so by themselves. Plus, even if they are customizing their application in similar manner as `libtorchaudio`, they can pick arbitral names for their compiler definitions. There is no obligation that they have to pick the same definition. So I do not think they need to be public. If we want to make them public, then I think we need to prefix the compiler definition with `LIBTORCHAUDIO_` to reduce the likelihood of name collision.